Fix: Corrected compatibility level returned by sb_helpdb

deepakshi-mittal opened the pull request:
Task: BABEL-4029

### Description
sp_helpdb will show 120 as well for compatibility level same as Sys.databases and sys.sysdatabases' result.


### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

deepakshi-mittal committed:
Fix: Corrected compatibility level returned by sb_helpdb
Task: BABEL-4029

Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>

macwoody commented on contrib/babelfishpg_tsql/src/catalog.c:
```diff
@@ -417,7 +417,7 @@ babelfish_helpdb(PG_FUNCTION_ARGS)
 
         nulls[5] = 1;
-		nulls[6] = 1;
+		values[6] = UInt8GetDatum(120);
```
At minimum, you may need a test covering this change

rishabhtanwar29 committed:
Support database-level dump/restore (#1304)

Engine

Added support for dumping a single Babelfish logical database. Changes are
summarized in following bullet points:

    Added a new option --bbf-database-name in both pg_dumpall and pg_dump to
    let user provide the name of logical database to be dumped.
    Fetch all the physical schema names corresponding to the given logical database
    and add all of them into schema_include_patterns list. With this, existing dump
    logic will make sure to only dump all these schemas as well as all the objects
    contained by these schemas.
    Function bbf_selectDumpableTableData takes care of explicitly marking all
    Babelfish catalog table data to be dumped.
    We need to selectively dump the Babelfish catalog table data corresponding to
    given logical database which is being handled by function getCursorForBbfCatalogTableData
    for COLUMN-INSERT and fixCopyCommand for COPY. These function prepares
    query using joins and conditions to filter the data.
    We skip dumping the dbid column as it might conflict with the existing database's dbid
    so we generate the new dbid during restore. It has been handled for both COLUMN-INSERT
    and COPY options:
    a. COPY: Implemented a hook fill_missing_values_in_copyfrom_hook which will use the
    sequence to generate the new dbid.
    b. COLUMN-INSERT: Used existing pre_parse_analyze_hook hook to handle this case.
    Added a new file dumpall_babel_utils.c to implement Babelfish specific logic to
    selectively dump users. New file is needed since existing dump_babel_utils.c can't
    be used with pg_dumpall utility.
    Logins do not belong to a single database so we will not be dumping the logins so it and
    we will only dump default users of the database (dbo, db_owner and db_guest).
    Due to this CREATE and INSERT might fail for certain logins but it should be fine.
    Current implementation does not dump foreign servers and foreign data wrappers as
    these are global objects and does not belong to a single logical database.
    Note: A logical database with certain migration mode (single-db/multi-db) can only be
    restored on the Babelfish server with the same migration mode. Which means a single-db
    database can be restored only on single-db Babelfish server, same thing applies to multi-db.
    Although the builtin databases like master can be restored on any kind of server.

Extension

    Modified existing instance dump-restore github action to work with database-level
    dump-restore as well. We can use configuration file to specify the logical database name.
    Currently we are testing with master database for which a directory has been in upgrade
    folder.
    Implemented fill_missing_values_in_copyfrom_hook to handle the case of generating
    new dbid for COPY. Handled the same for COLUMN-INSERT in pre_parse_analyze_hook.


Signed-off-by: Rishabh Tanwar <ritanwar@amazon.com>
Co-authored-by: Rishabh Tanwar <ritanwar@amazon.com>

deepakshi-mittal committed:
Added test cases for compatibility level
Task: BABEL-4029

Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>

deepakshi-mittal committed:
Modify test case

Task:BABEL-4029
Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>
